src/Datagrid/Pager.php

```diff
@@ -417,7 +417,7 @@ public function isFirstPage()
      */
     public function isLastPage()
     {
-        return $this->page === $this->lastPage;
+        return $this->page === (int) $this->lastPage;
```

Comment:
> Variable `lastPage` can be float.

What? How? Shouldn't we do the cast in `setLastPage`?

Reply:
maybe in a file `sonata-project/admin-bundle/src/Datagrid/Pager.php`

method `setLastPage`

and in `sonata-project/datagrid-bundle/src/Pager/BasePager.php`

Reply:
You are right, @greg0ire and @axzx. Initially, I thought that somewhere this value for a path might be overwritten, so for safety I would cast it in comparison. Corrected now.

Reply:
I think the right fix would be to cast to int _before_ calling `setLastPage`, wherever it is called (apparently, not in this library?). In the next major version of Sonata, we will add an `int` type hint, that will make the calling code crash if it is using `strict_types`. Let's fix the calling code.